Reporting Dashboard docs (#3341)
astronomer/docs: astro/reporting-dashboard.md

By examining your data at the Organization level, you can identify Deployments with large numbers of failing tasks by looking at the graphs or charts for outliers. Then, you can filter by time period to see if there have been similar events in the past and when.

![The main section of the Organization Overview dashboard, showing Deployments and task counts over time as a bar chart](/img/docs/dash-organization-overview.png)

## Deployment Detail

The **Deployment Detail** dashboard shows a table of all Deployments in your Organization, indexed by Workspace name. For each Deployment, the dashboard shows your Deployment configurations as well as use metrics like task run counts, DAG run counts, and operator counts.

Use this dashboard to check that your Deployment has the appropriate resources based on the number of DAGs it runs. You can also use this dashboard to check whether a Deployment's Astro Runtime version is currently supported.

![View data about Deployment health and DAG success.](/img/docs/dash-deployment-detail.png)

## DAG run detail

> This dashboard shows the average and median ending time of a specific DAG run, aggregated by a chosen time period, month, week or day. You can use this view to visually inspect when particular DAGs complete their first run, which allows you to see trends over time. This means you can make changes to your resources, such as Deployment settings or DAG code, and then check their performance over time, allowig you to highlight the impact of those changes.

Reviewer: ```suggestion
This dashboard shows the average and median ending time of a specific DAG run, aggregated by a chosen time period, month, week or day. You can use this view to visually inspect when particular DAGs complete their first run, which allows you to see trends over time. Check this dashboard when you make changes to Deployment resources or DAG code to track their effects on performance over time.
```

---
sidebar_label: 'Reporting dashboard'
title: "Reporting dashboard"
id: reporting-dashboard
description: "View information about your Organization, Deployments, DAGs, and costs."
---

:::warning

This feature is in [Private Preview](https://docs.astronomer.io/astro/feature-previews). Please reach out to your customer success manager to enable this feature.

:::

> Astro provides [a Dashboard](https://cloud.astronomer.io/dashboards) with a consolidated view of important metrics about your Organization's use of Astro. This includes data about your Deployment performance, DAG statistics, DAG SLA, operator use, and cost breakdown.

Reviewer 2: Add sentence about "why" you use these dashboards or what they can help you understand

## Access reporting Dashboards

You can find the Dashboards in your **Organization settings** menu, by clicking **Dashboards**.

> Only users with **Organization Owners** and **Organization Admins** [user permissions](user-permissions.md#organization-roles) can access reporting Dashboards.

Reviewer 2: Consider just adding the "how to access" at the bottom of the overview section since it's so short

> ## Export data from reporting Dashboards

Reviewer 2: I'd move this down to the bottom

You can set up your reporting Dashboard to email you data reports on a given time interval or as an alert when certain criteria are met. Or, you can export your data in a one-time action.

1. Hover your cursor anywhere in the Dashboard window or over a specific data element, to see the detailed menu view. This menu includes **Filters**, the option to **Maximize element**, or **Expand menu** further.

2. Expand the menu, and then you can choose
    - **Sort** the x-axis data
    - **Alert when...** to define criterial when Astro will send you a data export of the element or dashboard
    - **Export** to download the data in a file format of your choosing, like `.csv`.
    - **Refresh data** to reload the data you're viewing in the Dashboard or element.

## Organization Explorer

The **Organization Explorer** tab provides at-a-glance summaries about activity in your Organization's Deployment and DAGs in your entire Organization, or filtered by Workspace or Deployment name.

> This tab allows you to quickly inspect the activity and performance of your Deployments, and compare them to one another which means you can quickly identify any Deployments, DAGs, or tasks that have had recent behavior changes or are performing in an unexpected way. Hovering your cursor over any of the charts brings up a detailed view of the data, indexed by date.

Reviewer 2: Run on sentence?

> All Dashboards include sharing how recently the data was updated in the **Data available through**.

Reviewer 2: Capital "D" or not? Or are we using interchangeably with `report`?

* **Deployment name** - Choose which Deployments' data you want to view. By default, if you don't select any Deployment, the report shows the aggregated view of all Deployments.
* **Display by** - Select the time interval you want to view over. This view can be as granular as the **Day** or view data aggregated by **Week** or **Month**.

### Organization explorer data

In the Dashboard, you can take a detailed view of the following different reporting elements. Each has its own section where you can filter the data within the element, export reports, or set up an email alert.

* **Deployments** - The number of Deployments in a Workspace, indexed by time.
* **Task Counts by Success/Failure** - Segmented bar chart showing the counts of successful and failed tasks in the selected Workspace, indexed by time.
* **Active DAG counts** -
* **Task Counts by Workspace** - A line graph of the count of tasks in selected Workspaces over time. If you select more
* **Workspace summary in last 30 Days**
    - Top 10 Workspaces by code push count
    - Top 10 Workspaces by active DAG count
    - Top 10 Workspaces by unique operator count

## Deployment Detail

> The **Deployment Detail** report shows a table of all your non-deleted Deployments in your Organization, indexed by Workspace Name. You can filter this table by **Workspace Name** or **Deployment Name** Additionally, the task counts, DAG counts, and unique operator counts are shown for the latest completed month.

Reviewer: ```suggestion
The **Deployment Detail** report shows a table of all Deployments in your Organization, indexed by Workspace Name. For each Deployment, the dashboard shows your Deployment configurations as awell as usage metrics including task run counts, DAG run counts, and operator counts.
```
- Are there any dashboards where you can view deleted Deployments? If not, then the distinction here seems unnecessary
- Are these task RUN counts or just task counts? I assume it's the former if it's a usage dashboard. But we should confirm

> This report view also highlights if you have any Deployments that use a version of the Astro Runtime that are either unsupported or are expected to reach end of maintenance in the next six months.

Reviewer: ```suggestion
Use this dashboard to check that your Deployment has the appropriate resources based on the number of DAGs it runs. You can also use this dashboard to check whether your Astro Runtime version is in support or needs to be upgraded.
```

## DAG SLA

> If you have a service level agreement (SLA) where your DAGs need to complete by a certain consistent time, you can use this view to compare DAG completion times with an SLA time. This view can help you report the frequency your Organization meets their SLA obligations.

Reviewer: ```suggestion
The DAG SLA dashboard allows you to simulate a service level agreement (SLA) for when a DAG should complete and then compare your DAG runs against the SLA. Use this dashboard to ensure that your Organization meets your SLA obligations, or to create and alert on new SLAs.

To use this dashboard, you must first define an SLA and the DAGs which should be compared against the SLA.
```

Reviewer: This one is tricky because you can actually configure an SLA at the DAG or task level in code, and this dashboard is only a simulation of those SLAs. It has no affect/ connection to real SLAs that users might have configured. 

That might be worth clarifying, and we also might want to ask product whether we should recommend users "Alert on" a metric SLA instead of configuring their own SLA in DAGs or Astro Alerts.

1. Choose the **DAG ID** that you want to check DAG run completion times for. If you have a large number of DAGs in your Organization, you can first filter by **Workspace Name** and **Deployment Name** before choosing your DAG ID.
2. Select the time interval over which you want to look at the DAG data.
3. Define the **SLA Time (UTC)**
4. Choose the time interval you want to aggregate the data with by choosing **Day**, **Week**, or **Month** in **Display by**.

> The report generates a line graph that shows the **Median End Time** of the DAG, the **Average End Time**, and the **SLA** time you defined.

Reviewer: ```suggestion
The report generates a line graph that shows the **Median End Time** of the DAG, the **Average End Time**, compared to the **SLA** time you defined.
```

> ## Billing Details (Hosted Only)

Reviewer: Any reason we chose a different title from the dashboard name?

Author: It got changed!

> In the **Cost Breakdown** tab of your Dashboard, you can find detailed information regarding your accumulated cost per time period, depending on the Workspace Name and Deployment.

Reviewer: ```suggestion
In the **Cost Breakdown** tab of your Dashboard, you can find detailed information about how much you spend on Astro over time for each of your Workspaces and Deployments. Use this data alongside your other dashboards to identify the biggest opportunities for cost reduction in your Organization.
```

* **Date**
* **Workspace Name**: Choose which Workspaces' details you want to examine in detail.
* **Deployment Name**: Choose which Deployments' details you want to examine in detail.

### Cost breakdown data details

- **Total spend** - The total cost per time period.
- **Deployment spend**- The cost per time period related to your Deployment scheduler configuration.
- **Compute spend** - The cost per time period related to your Worker runtime.
- **Workspaces** - The Number of Workspaces in your Organization.
- **Deployments** - The Number of Deployments in your Organization
- **Workspace Spend** - A segmented bar chart that shows the **Deployment** and **Compute** spend amount, organized by Workspace.
- **Spend for all Workspaces** - A segmented bar chart that shows the aggregated cost over time, with each segment representing the cost of a specific Workspace.
- **Spend by Workspace Deployments** - <!--needs feedback from data team-->
- **Detailed Spend for Deployments** - This detailed table describes the accumulated cost in terms of **Billable metrics**, which are the same metrics used in your invoices.

## Operator usage

> You can examine how your Deployments and Workspaces use Operators, and the frequency that their tasks succeed and fail.

Reviewer: ```suggestion
You can examine how your Deployments and Workspaces use Operators, and the frequency that their tasks succeed and fail. Use this data to identify types of operators that could be replaced with more efficient alternatives, or to find operators that fail more than expected. 
```

* **Workspace Operator Usage by Task/Distinct DAG Count** - This view shows you the most used Workspaces based on total Task Count and Distinct DAGs. You can then expand each row to find the most frequently used Operators in each Deployment in that Workspace. This data answers the question, *Which workspace is using the selected operators the most?*
* **Task Usage By Period, Failed/Successful** - An aggregate view of the number of tasks that failed or succeeded in the selected Workspace and Deployment over the defined time period.
* **Task Usage By Period, Workspace** - An aggregate view of the number of tasks by Workspace over the defined time period.

## Export data from reporting Dashboards

You can set up your reporting Dashboard to email you data reports on a given time interval or as an alert when certain criteria are met. Or, you can export your data in a one-time action.

1. Hover your cursor anywhere in the Dashboard window or over a specific data element, to see the detailed menu view. This menu includes **Filters**, the option to **Maximize element**, or **Expand menu** further.

2. Expand the menu, and then you can choose
    - **Sort** the x-axis data

>     - **Alert when...** to define criterial when Astro will send you a data export of the element or dashboard

Author: It does sound different, but the actual action is a data export like the others, it's just done conditionally I think?

>     - **Alert when...** to define criterial when Astro will send you a data export of the element or dashboard

Reviewer: I think the use case is different. A user might complete a manual export when they want data for a report or a presentation. But a user setting up an alert is doing so because they want to be alerted when something is wrong in the system, and they want the data presented to them immediately when said thing happens. 

Given the discrepancy in use cases, I don't think it hurts to have these documented as separate topics. We should also see whether the product team expects users to configure these alerts as part of their overall alerting infrastructure - for example, do we want users to replace their Airflow code level SLAs with our own SLA dashboard alerts?

>     - **Alert when...** to define criterial when Astro will send you a data export of the element or dashboard

Author: @jwitz right now there's some work going on with the alerting/exporting functionality. How do you feel about me commenting this section out until that gets resolved so that we can get the page out and keep following with additional functionality?

>     - **Alert when...** to define criterial when Astro will send you a data export of the element or dashboard

Reviewer: Sounds good!

---
sidebar_label: 'Reporting dashboards'
title: "View metrics across your Organization with Astro reporting dashboards"
id: reporting-dashboard
description: "View information about your Organization, Deployments, DAGs, and costs."
---

import HostedBadge from '@site/src/components/HostedBadge';

:::warning

This feature is in [Private Preview](https://docs.astronomer.io/astro/feature-previews). Please reach out to your customer success manager to enable this feature.

:::

> Astro provides a collection of dashboards that include important metrics about your Organization's use of Astro. These dashboards include:

Reviewer: Yes agreed on all fronts! Let's keep "dashboards" and also use "elements"

- Operator Use

Dashboards help you quickly identify opportunities to optimize how your team uses Airflow. For example, you can use the Deployment Detail page to identify unexpected DAG behaviors, like unusually high compute use, and then check the **Cost Breakdown** dashboard to identify any associated costs incurred by that behavior.

To view Astro dashboards, click your Workspace name in the upper left corner of the Cloud UI, click **Organization Settings**, then click **Dashboards**. You can also access this page directly at [https://cloud.astronomer.io/dashboards](https://cloud.astronomer.io/dashboards). 

Use the tabs at the bottom of the Cloud UI to switch between dashboards. Each dashboard shows the last time that it was updated in the **Data available through** section.

:::info

Only users with **Organization Billing Admin** [user permissions](user-permissions.md#organization-roles) can access reporting dashboards.

:::

## Organization Explorer

The **Organization Explorer** dashboard provides at-a-glance summaries about activity across your Organization's Deployments. You can also filter to view summaries for specific Workspaces or Deployments.

> This tab allows you compare the activity and performances of Deployments to one another. You can identify Deployments, DAGs, or tasks that have had recent behavior changes or are performing in an unexpected way. For example, you can filter the data shown by time period, Workspace name, or Deployment name, to view data such as the number of successful or failed tasks within Workspaces or Deployments. Hovering your cursor over any of the charts brings up a detailed view of the data, indexed by date.

Reviewer: ```suggestion
This tab allows you to compare the activity and performances of Deployments to one another. You can identify Deployments, DAGs, or tasks that have had recent behavior changes or are performing in an unexpected way. For example, you can filter the data shown by time period, Workspace name, or Deployment name, to view data such as the number of successful or failed tasks within Workspaces or Deployments. Hovering your cursor over any of the charts brings up a detailed view of the data, indexed by date.
```

By examining your data at the Organization level, you can identify Deployments with large numbers of failing tasks by looking at the graphs or charts for outliers. Then, you can filter by time period to see if there have been similar events in the past and when.

![The main section of the Organization Overview dashboard, showing Deployments and task counts over time as a bar chart](/img/docs/dash-organization-overview.png)

## Deployment Detail

The **Deployment Detail** dashboard shows a table of all Deployments in your Organization, indexed by Workspace name. For each Deployment, the dashboard shows your Deployment configurations as well as use metrics like task run counts, DAG run counts, and operator counts.

Use this dashboard to check that your Deployment has the appropriate resources based on the number of DAGs it runs. You can also use this dashboard to check whether a Deployment's Astro Runtime version is currently supported.

![View data about Deployment health and DAG success.](/img/docs/dash-deployment-detail.png)

> ## DAG SLA

Author: @jwitz this is another section that might get temporarily commented out until we get UI updates aligned with docs updates.

---

> sidebar_label: 'Reporting dashboard'

Reviewer: ```suggestion
sidebar_label: 'Reporting dashboards'
```

> title: "Reporting dashboard"

Reviewer: ```suggestion
title: "View metrics across your Organization with Astro reporting dashboards"
```

:::info

Only users with **Organization Billing Admin** [user permissions](user-permissions.md#organization-roles) can access reporting dashboards.

:::

## Organization Explorer

The **Organization Explorer** tab provides at-a-glance summaries about activity across your Organization's Deployments. You can also filter to view summaries for specific Workspaces or Deployments.

This tab allows you compare the activity and performances of Deployments to one another. You can identify Deployments, DAGs, or tasks that have had recent behavior changes or are performing in an unexpected way. For example, you can filter the data shown by time period, Workspace name, or Deployment name, to view data such as the number of successful or failed tasks within Workspaces or Deployments. Hovering your cursor over any of the charts brings up a detailed view of the data, indexed by date.

> By examining your data at the Organization level, you can quickly identify Deployments with large numbers of failing tasks by simply looking at the graphs or charts for outliers. Then, you can filter by time period to see if there have been similar events in the past and when.

Reviewer: ```suggestion
By examining your data at the Organization level, you can identify Deployments with large numbers of failing tasks by looking at the graphs or charts for outliers. Then, you can filter by time period to see if there have been similar events in the past and when.
```

> ![Quickly examine summaries about your Organization's DAG activity at the DAG and Workspace level.](/img/docs/dash-organization-overview.png)

Reviewer: ```suggestion
![The main section of the Organization Overview dashboard, showing Deployments and task counts over time as a bar chart](/img/docs/dash-organization-overview.png)
```

## Deployment Detail

> The **Deployment Detail** dashboard shows a table of all Deployments in your Organization, indexed by Workspace Name. For each Deployment, the dashboard shows your Deployment configurations as well as use metrics like task run counts, DAG run counts, and operator counts.

Reviewer: ```suggestion
The **Deployment Detail** dashboard shows a table of all Deployments in your Organization, indexed by Workspace name. For each Deployment, the dashboard shows your Deployment configurations as well as use metrics like task run counts, DAG run counts, and operator counts.
```

> Use this dashboard to check that your Deployment has the appropriate resources based on the number of DAGs it runs. You can also use this dashboard to check whether your Astro Runtime version is currently supported or if you need to upgrade it.

Reviewer: ```suggestion
Use this dashboard to check that your Deployment has the appropriate resources based on the number of DAGs it runs. You can also use this dashboard to check whether a Deployment's Astro Runtime version is currently supported.
```

![View data about Deployment health and DAG success.](/img/docs/dash-deployment-detail.png)

## DAG SLA

> The DAG SLA dashboard allows you to simulate a service level agreement (SLA) for when a DAG should complete and then compare how frequently your DAG runs complete within the SLA. Use this dashboard to check how frequently your Organization meets your SLA obligations, or to create an alert on new SLAs.

Reviewer: ```suggestion
The DAG SLA dashboard allows you to simulate a service level agreement (SLA) for when a DAG should complete and then compare how frequently your DAG runs complete within the SLA. Use this dashboard to check how frequently your Organization meets your SLA obligations, or to create an alert when specific SLAs are breached.
```

![Examine how frequently DAGs meet your SLA.](/img/docs/dash-DAG-SLA.png)

> ### Set up an SLA

Reviewer: ```suggestion
### Create an SLA simulation
```

> To create an SLA, you must first define an SLA and then select the DAGs you want to compare it to.

Reviewer: ```suggestion
To simulate an SLA, you must first define an SLA and then apply it to specific DAGs.
```

1. Choose the **DAG ID** that you want to check DAG run completion times for. If you have a large number of DAGs in your Organization, you can first filter by **Workspace Name** and **Deployment Name** before choosing your DAG ID.
2. Select the time interval over which you want to look at the DAG data.
3. Define the **SLA Time (UTC)**
4. Choose the time interval you want to aggregate the data with by choosing **Day**, **Week**, or **Month** in **Display by**.

The report generates a line graph that shows the **Median End Time** of the DAG, the **Average End Time**, compared to the **SLA** time you defined.

## Cost Breakdown
<HostedBadge/>

> In the **Cost Breakdown** tab of your Dashboard, you can find detailed information about how much you spend on Astro over time for each of your Workspaces and Deployments. Use this data alongside your other dashboards to identify the biggest opportunities for cost reduction in your Organization.

Reviewer: ```suggestion
In the **Cost Breakdown** dashboard, you can find detailed information about how much you spend on Astro over time for each of your Workspaces and Deployments. Use this data alongside your other dashboards to identify the biggest opportunities for cost reduction in your Organization.
```

![View the cost of your different Astro resources.](/img/docs/dash-cost-breakdown.png)

## Operator use

> You can examine how your Deployments and Workspaces use Operators, and the frequency that their tasks succeed and fail. Use this data to identify types of operators that could be replaced with more efficient alternatives, or to find operators that fail more than expected. This dashboard provides data to answer the questions, *What are the top operators used across my organization?* and *Which workspace is using the selected operators the most?*.

Reviewer: ```suggestion
The **Operator Use** dashboard shows how your Deployments and Workspaces use Operators, as well as how often tasks succeed and fail when using specific operators. Use this data to identify types of operators that could be replaced with more efficient alternatives, or to find operators that fail more than expected. This dashboard provides data to answer the questions, *What are the top operators used across my organization?* and *Which workspace is using the selected operators the most?*.
```

> 
> You can filter your data, export reports, or set up an email alert, which sends a copy of the data to an email address under certain circumstances.

Reviewer: ```suggestion
```
Seems like a general description unrelated to the dashboard
